### PR TITLE
CS-38921 [Branch support for export to csv] - Merge into development  

### PR DESCRIPTION
**Pull request by akashbhosale97**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=g32m4qR)

Jira ticket : CS-38921
Version bump : 1.3.8 --> 1.3.9

### CODE REVIEW

File: packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js [hunk 1]
```diff
@@ -110,7 +117,43 @@ class ExportToCsvCommand extends Command {
...
+            if (branchUid) {
+              try {
+                const isBranchExist = await stackAPIClient.branch(branchUid).fetch();
+                if (isBranchExist && typeof isBranchExist === 'object') {
+                  stack.branch_uid = branchUid;
+                  stackAPIClient = this.getStackClient(managementAPIClient, stack);
+                }
+              } catch (error) {
+                switch (error.errorCode) {
+                  case 905: {
```
Comment (netrajpatel) [inline packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js:130]: @akashbhosale97 Are we still managing this error handling through the response’s error codes?

File: packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js [hunk 2]
```diff
@@ -110,7 +117,43 @@ class ExportToCsvCommand extends Command {
...
+                    cliux.error(
+                      'The branch you entered does not exist in the selected stack. Please enter a valid branch.',
+                    );
+                    process.exit(1);
+                  }
+                  case 412: {
```
Comment (netrajpatel) [inline packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js:136]: @akashbhosale97 Are we still managing this error handling through the response’s error codes?
This was supposed to be handled through the branch feature check utility right?

**Review by netrajpatel — CHANGES_REQUESTED** [review]

File: packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js (hunk 1 above)
Comment (akashbhosale97) [reply packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js]: changed it to use utility

File: packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js (hunk 2 above)
Comment (akashbhosale97) [reply packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js]: changed it to use utility

**Comment by akashbhosale97** [comment]
> @akashbhosale97 Can you please add the ticket number and version bump info in PR description? Also, this branch needs to be updated with latest development changes.

added those in the PR description

File: packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js [hunk 3]
```diff
@@ -110,7 +124,34 @@ class ExportToCsvCommand extends Command {
...
+              try {
+                const branchExists = await doesBranchExist(stackAPIClient, branchUid);
+                if (branchExists && branchExists.errorCode) {
+                  throw new Error(branchExists.errorMessage);
+                }
+                stack.branch_uid = branchUid;
+                stackAPIClient = this.getStackClient(managementAPIClient, stack);
+              } catch (error) {
+                if (error.message || error.errorMessage) {
+                  this.log(util.formatError(error));
```
Comment (netrajpatel) [inline packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js:138]: @akashbhosale97 Error messages should be in red

File: packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js [hunk 4]
```diff
@@ -1,5 +1,12 @@
 const { Command } = require('@contentstack/cli-command');
-const { configHandler, managementSDKClient, flags, isAuthenticated } = require('@contentstack/cli-utilities');
+const {
+  configHandler,
+  managementSDKClient,
+  flags,
+  isAuthenticated,
+  cliux,
```
Comment (netrajpatel) [inline packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js:7]: @akashbhosale97 If cliux is not used anywhere, can you remove this?

File: packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js [hunk 5]
```diff
@@ -110,7 +124,34 @@ class ExportToCsvCommand extends Command {
...
+            if (branchUid) {
+              try {
+                const branchExists = await doesBranchExist(stackAPIClient, branchUid);
+                if (branchExists && branchExists.errorCode) {
+                  throw new Error(branchExists.errorMessage);
+                }
+                stack.branch_uid = branchUid;
+                stackAPIClient = this.getStackClient(managementAPIClient, stack);
+              } catch (error) {
+                if (error.message || error.errorMessage) {
```
Comment (netrajpatel) [inline packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js:137]: There is a Sonar-lint issue on this line.

File: packages/contentstack-export-to-csv/src/util/index.js [hunk 6]
```diff
@@ -126,6 +126,26 @@ function chooseStack(managementAPIClient, orgUid) {
...
+      {
+        type: 'list',
+        name: 'branch',
+        message: 'Choose a Branch',
+        choices: branchesArray,
+      },
+    ];
+    return await inquirer.prompt(_chooseBranch);
+  } catch (err) {
+    console.log(err);
```
Comment (netrajpatel) [inline packages/contentstack-export-to-csv/src/util/index.js:145]: @akashbhosale97 Remove console statements and use Cliux.

**Review by netrajpatel — CHANGES_REQUESTED** [review]

File: packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js (hunk 5 above)
Comment (akashbhosale97) [reply packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js:137]: fixed it

**Review by netrajpatel — APPROVED** [review]

**Pull request merged**